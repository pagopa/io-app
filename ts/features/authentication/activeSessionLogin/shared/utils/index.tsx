import I18n from "i18next";
import { Millisecond } from "@pagopa/ts-commons/lib/units";
import { Platform } from "react-native";
import { CieAuthenticationErrorReason } from "../../../login/cie/store/actions/cie";
import { ReadingState } from "../../../login/cie/components/CieCardReadingAnimation";

// the timeout we sleep until move to consent form screen when authentication goes well
export const WAIT_TIMEOUT_NAVIGATION = 1700 as Millisecond;
export const WAIT_TIMEOUT_NAVIGATION_ACCESSIBILITY = 5000 as Millisecond;
export const VIBRATION = 100 as Millisecond;
export const accessibityTimeout = 100 as Millisecond;

// A subset of Cie Events (errors) which is of interest to analytics
export const analyticActions = new Map<CieAuthenticationErrorReason, string>([
  // Reading interrupted before the sdk complete the reading
  ["Transmission Error", I18n.t("authentication.cie.card.error.onTagLost")],
  ["ON_TAG_LOST", I18n.t("authentication.cie.card.error.onTagLost")],
  [
    "TAG_ERROR_NFC_NOT_SUPPORTED",
    I18n.t("authentication.cie.card.error.unknownCardContent")
  ],
  [
    "ON_TAG_DISCOVERED_NOT_CIE",
    I18n.t("authentication.cie.card.error.unknownCardContent")
  ],
  ["PIN Locked", I18n.t("authentication.cie.card.error.generic")],
  ["ON_CARD_PIN_LOCKED", I18n.t("authentication.cie.card.error.generic")],
  ["ON_PIN_ERROR", I18n.t("authentication.cie.card.error.tryAgain")],
  ["PIN_INPUT_ERROR", ""],
  ["CERTIFICATE_EXPIRED", I18n.t("authentication.cie.card.error.generic")],
  ["CERTIFICATE_REVOKED", I18n.t("authentication.cie.card.error.generic")],
  ["AUTHENTICATION_ERROR", I18n.t("authentication.cie.card.error.generic")],
  [
    "EXTENDED_APDU_NOT_SUPPORTED",
    I18n.t("authentication.cie.nfc.apduNotSupported")
  ],
  [
    "ON_NO_INTERNET_CONNECTION",
    I18n.t("authentication.cie.card.error.tryAgain")
  ],
  ["STOP_NFC_ERROR", ""],
  ["START_NFC_ERROR", ""]
]);

export type TextForState = {
  title: string;
  subtitle?: string;
  content: string;
};

// some texts changes depending on current running Platform
export const getTextForState = (
  state: ReadingState.waiting_card | ReadingState.error,
  errorMessage: string = ""
): TextForState => {
  const texts: Record<
    ReadingState.waiting_card | ReadingState.error,
    TextForState
  > = Platform.select({
    ios: {
      [ReadingState.waiting_card]: {
        title: I18n.t("authentication.cie.card.titleiOS"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeaderiOS"),
        // the native alert hides the screen content and shows a message it self
        content: ""
      },
      [ReadingState.error]: {
        title: I18n.t("authentication.cie.card.error.readerCardLostTitle"),
        subtitle: "",
        // the native alert hides the screen content and shows a message it self
        content: ""
      },
      [ReadingState.reading]: {
        title: I18n.t("authentication.cie.card.titleiOS"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeaderiOS"),
        // the native alert hides the screen content and shows a message it self
        content: ""
      }
    },
    default: {
      [ReadingState.waiting_card]: {
        title: I18n.t("authentication.cie.card.title"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeader"),
        content: I18n.t("authentication.cie.card.layCardMessageFooter")
      },
      [ReadingState.error]: {
        title: I18n.t("authentication.cie.card.error.readerCardLostTitle"),
        subtitle: I18n.t("authentication.cie.card.error.onTagLost"),
        content: errorMessage
      }
    }
  });
  return texts[state];
};
