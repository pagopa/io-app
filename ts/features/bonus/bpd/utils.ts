import { PatchedWalletV2 } from "../../../types/pagopa";

/**
 * Return true if bpd is enabled for the wallet (aka payment method)
 * TODO: replace string with enum
 * @param wallet
 */
export const hasBpdCapability = (wallet: PatchedWalletV2) =>
  wallet.enableableFunctions.includes("BPD");
