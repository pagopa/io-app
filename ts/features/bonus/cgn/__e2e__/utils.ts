import fetch from "node-fetch";
import { e2eWaitRenderTimeout } from "../../../../__e2e__/config";
import I18n from "../../../../i18n";

export const activateCGNBonusSuccess = async () => {
  const startActivationCta = element(by.id("activate-bonus-button"));
  await waitFor(startActivationCta)
    .toBeVisible()
    .withTimeout(e2eWaitRenderTimeout);
  await startActivationCta.tap();
  await waitFor(element(by.id("cgnConfirmButtonTestId")))
    .toBeVisible()
    .withTimeout(e2eWaitRenderTimeout);

  // We should unsubscribe after the activation, in order to allow the next test run (we can remove this part when we will have the possibility
  // to reset the dev-server with an API command.

  // Go to bonus details
  await element(by.id("cgnConfirmButtonTestId")).tap();

  // wait for unsubscribe cta
  const scrollView = element(by.id("CGNCardDetailsScrollView"));

  // The section has a loading spinner on top of
  // everything so we must wait for it to disappear
  await waitFor(scrollView)
    .toBeVisible()
    .withTimeout(2 * e2eWaitRenderTimeout);

  // make sure to scroll to bottom, otherwise in small devices the element will not be visible nor tappable
  await scrollView.scrollTo("bottom");

  const unsubscribeCgnCta = element(by.id("cgnDeactivateBonusTestId"));
  await waitFor(unsubscribeCgnCta)
    .toBeVisible()
    .withTimeout(e2eWaitRenderTimeout);
  // unsubscribe
  await unsubscribeCgnCta.tap();

  // confirm alert
  const alertCTA = element(by.text(I18n.t("global.buttons.deactivate")));
  await waitFor(alertCTA).toBeVisible().withTimeout(e2eWaitRenderTimeout);
  await alertCTA.tap();
};

export const deactivateCGNCardIfNeeded = async () => {
  // This is needed since an E2E can fail at any moment. If it does so
  // after the card has been activated, all subsequent retries and tests
  // will fail, since the card is already activated. Be aware that this
  // endpoint is not authenticated (or, better said, when the authorization
  // header is not set, the fastLoginMiddleware, in the dev-server, skips
  // any authentication check)
  const cgnDeactivationEndpoint = "http://127.0.0.1:3000/api/v1/cgn/delete";
  // eslint-disable-next-line functional/no-let
  let responseCode = -1;
  try {
    const response = await fetch(cgnDeactivationEndpoint, {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json"
      }
    });
    responseCode = response.status;
  } catch (error) {
    // We are not really interested about the error
    // eslint-disable-next-line no-console
    console.log(`deactivateCGNCardIfNeeded: ${error}`);
  }

  if (responseCode !== 201 && responseCode !== 403) {
    throw new Error(
      `Invalid response code from CGN deactivateion (${responseCode}), check the CGN deactivation endpoint (${cgnDeactivationEndpoint}) and the dev-server configuration for CGN deactivation`
    );
  }
};
