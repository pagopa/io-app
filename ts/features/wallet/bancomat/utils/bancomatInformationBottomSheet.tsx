import * as React from "react";
import { View } from "react-native";
import { VSpacer } from "@pagopa/io-app-design-system";
import I18n from "../../../../i18n";
import InternationalCircuitIconsBar from "../../../../components/wallet/InternationalCircuitIconsBar";
import ButtonDefaultOpacity from "../../../../components/ButtonDefaultOpacity";
import { Label } from "../../../../components/core/typography/Label";
import { H4 } from "../../../../components/core/typography/H4";
import { useLegacyIOBottomSheetModal } from "../../../../utils/hooks/bottomSheet";

/**
 * A bottomsheet that display generic information on bancomat and a cta to start the onboarding of a new
 * payment method.
 * This will be also visualized inside a bottomsheet after an addition of a new bancomat
 */
export default (onAdd?: () => void) => {
  const { present, bottomSheet, dismiss } = useLegacyIOBottomSheetModal(
    <View>
      <InternationalCircuitIconsBar />
      <VSpacer size={16} />
      <H4 weight={"Regular"} color={"bluegreyDark"}>
        {I18n.t("wallet.bancomat.details.debit.body")}
      </H4>
      <VSpacer size={16} />
      <ButtonDefaultOpacity
        primary={true}
        block={true}
        bordered={true}
        onPress={() => {
          onAdd?.();
          dismiss();
        }}
        onPressWithGestureHandler={true}
      >
        <Label>{I18n.t("wallet.bancomat.details.debit.addCta")}</Label>
      </ButtonDefaultOpacity>
    </View>,
    I18n.t("wallet.bancomat.details.debit.title"),
    385
  );
  return {
    present,
    bottomSheet,
    dismiss
  };
};
