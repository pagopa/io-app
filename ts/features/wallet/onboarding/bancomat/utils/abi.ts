import { Abi } from "../../../../../../definitions/pagopa/walletv2/Abi";

/**
 * This function aims to order the list of Abis returned by the PM in alphabetical order based on his name
 * @param abis the abi list returned from the PM
 */
export const sortAbiByName = (abis: ReadonlyArray<Abi>) =>
  [...abis].sort((abi1: Abi, abi2: Abi) => {
    const abi1Name = abi1.name ?? "";
    const abi2Name = abi2.name ?? "";
    return abi1Name
      .toLocaleLowerCase()
      .localeCompare(abi2Name.toLocaleLowerCase());
  });
