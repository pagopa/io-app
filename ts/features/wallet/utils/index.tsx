import * as React from "react";
import { WalletCard, WalletCardCategoryFilter, WalletCardType } from "../types";
import { WalletCardBaseComponent } from "../components/WalletCardBaseComponent";
import { CgnWalletCard } from "../../bonus/cgn/components/CgnWalletCard";
import { IdPayWalletCard } from "../../idpay/wallet/components/IdPayWalletCard";
import { PaymentWalletCard } from "../../payments/wallet/components/PaymentWalletCard";
import { WalletCardSkeleton } from "../components/WalletCardSkeleton";
import { ItwCredentialWalletCard } from "../../itwallet/wallet/components/ItwCredentialWalletCard";
import { shouldRenderWalletCategorySelector } from "../store/selectors";
import { useIOSelector } from "../../../store/hooks";
import { GlobalState } from "../../../store/reducers/types";

/**
 * Wallet card component mapper which translates a WalletCardType to a
 * component to be rendered inside the wallet.
 * Component MUST be a WalletCardBaseComponent, which can be created
 * using {@see withWalletCardBaseComponent} HOC
 */
export const walletCardComponentMapper: Record<
  WalletCardType,
  WalletCardBaseComponent<any> | undefined
> = {
  cgn: CgnWalletCard,
  idPay: IdPayWalletCard,
  payment: PaymentWalletCard,
  itw: ItwCredentialWalletCard,
  placeholder: WalletCardSkeleton
};

/**
 * Function that renders a wallet card using the mapped component inside {@see walletCardComponentMapper}
 * @param card - The wallet card object to render
 * @param stacked - Whether the card is stacked or not
 * @returns The rendered card or null if the card is not found
 */
export const renderWalletCardFn = (
  card: WalletCard,
  stacked: boolean = false
) => {
  const { key, category, type, ...cardProps } = card;
  const Component = walletCardComponentMapper[type];
  return Component ? (
    <Component
      key={key}
      testID={`walletCardTestID_${category}_${type}_${key}`}
      cardProps={cardProps}
      isStacked={stacked}
    />
  ) : null;
};

/**
 * A higher-order component which renders a component only if the category filter matches the given category
 * @param category - The category to filter by
 * @param WrappedComponent - The component to render
 * @returns The component or null if the category filter does not match
 */
export const withWalletCategoryFilter =
  <P extends Record<string, unknown>>(
    category: WalletCardCategoryFilter,
    WrappedComponent: React.ComponentType<P>
  ) =>
  (props: P) => {
    const shouldRenderCategory = useIOSelector((state: GlobalState) =>
      shouldRenderWalletCategorySelector(state, category)
    );
    if (!shouldRenderCategory) {
      return null;
    }
    return <WrappedComponent {...props} />;
  };
