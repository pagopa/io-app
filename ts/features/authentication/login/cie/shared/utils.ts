import { Platform } from "react-native";
import { isDevEnv } from "../../../../../utils/environment";
import { SpidLevel } from "../utils";

export type WebViewLoginNavigationProps = {
  spidLevel: SpidLevel;
  isUat: boolean;
};

export const iOSUserAgent =
  "Mozilla/5.0 (iPhone; CPU iPhone OS 14_0_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1";

export const defaultUserAgent = Platform.select({
  ios: iOSUserAgent,
  default: undefined
});

export const originSchemasWhiteList = [
  "https://*",
  "iologin://*",
  ...(isDevEnv ? ["http://*"] : [])
];

export const WHITELISTED_DOMAINS = [
  "https://idserver.servizicie.interno.gov.it",
  "https://oidc.idserver.servizicie.interno.gov.it",
  "https://mtls.oidc.idserver.servizicie.interno.gov.it",
  "https://mtls.idserver.servizicie.interno.gov.it",
  "https://ios.idserver.servizicie.interno.gov.it",
  "https://ios.oidc.idserver.servizicie.interno.gov.it"
];

export type CieIdLoginProps = {
  spidLevel: SpidLevel;
  isUat: boolean;
};
