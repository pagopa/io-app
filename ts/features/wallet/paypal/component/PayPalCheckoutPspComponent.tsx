import * as React from "react";
import { ListItem, View } from "native-base";
import { StyleSheet } from "react-native";
import { H3 } from "../../../../components/core/typography/H3";
import { ImportoEuroCents } from "../../../../../definitions/backend/ImportoEuroCents";
import { H4 } from "../../../../components/core/typography/H4";
import { formatNumberCentsToAmount } from "../../../../utils/stringBuilder";
import { Label } from "../../../../components/core/typography/Label";
import IconFont from "../../../../components/ui/IconFont";
import { IOColors } from "../../../../components/core/variables/IOColors";
import I18n from "../../../../i18n";
import { Link } from "../../../../components/core/typography/Link";
import { openWebUrl } from "../../../../utils/url";

const styles = StyleSheet.create({
  column: {
    flex: 1,
    flexDirection: "column"
  },
  pspContainer: {
    alignItems: "flex-start",
    flexDirection: "column"
  },
  pspHeader: {
    flexDirection: "row"
  },
  rowCentered: { flex: 1, flexDirection: "row", alignItems: "center" }
});

type Props = {
  fee: ImportoEuroCents;
  pspName: string;
  privacyUrl?: string;
  onEditPress: () => void;
};

// temporary feature flag since this feature is still WIP (missing task to complete https://pagopa.atlassian.net/browse/IA-684?filter=10121)
export const editPspEnabled = false;
/**
 * this component shows the psp details (cost, name and privacy url) associated
 * to the current psp that is handling the payment addressed with Paypal
 * It also shows an edit label to navigate to the psp list to pick a new one
 * @constructor
 */
export const PayPalCheckoutPspComponent = (props: Props) => {
  const { privacyUrl } = props;
  return (
    <View style={styles.column}>
      <ListItem style={styles.pspContainer}>
        <View style={styles.pspHeader}>
          <IconFont
            name="io-transactions"
            size={24}
            color={IOColors.bluegrey}
          />
          <View hspacer small={true} />
          <H3 color={"bluegrey"}>
            {I18n.t("wallet.onboarding.paypal.paymentCheckout.transactionCost")}
          </H3>
        </View>
        <View spacer={true} />
        <View style={styles.rowCentered}>
          <View style={styles.column}>
            <H4>{formatNumberCentsToAmount(props.fee, true)}</H4>
            <Label color={"bluegrey"} weight={"Regular"}>
              {I18n.t("wallet.onboarding.paypal.paymentCheckout.pspHandler", {
                pspName: props.pspName
              })}
            </Label>
          </View>
          {editPspEnabled && (
            <Label onPress={props.onEditPress} testID={"editLabelTestID"}>
              {I18n.t("global.buttons.edit").toUpperCase()}
            </Label>
          )}
        </View>
      </ListItem>
      {privacyUrl && (
        <>
          <View spacer={true} />
          <Label color={"bluegrey"} weight={"Regular"}>
            {I18n.t(
              "wallet.onboarding.paypal.paymentCheckout.privacyDisclaimer"
            )}
          </Label>
          <Link onPress={() => openWebUrl(privacyUrl)}>
            {I18n.t("wallet.onboarding.paypal.paymentCheckout.privacyTerms")}
          </Link>
        </>
      )}
    </View>
  );
};
