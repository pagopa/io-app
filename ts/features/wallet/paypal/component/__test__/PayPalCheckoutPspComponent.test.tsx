import { render } from "@testing-library/react-native";
import React from "react";
import { constNull } from "fp-ts/lib/function";
import { PayPalCheckoutPspComponent } from "../PayPalCheckoutPspComponent";
import { ImportoEuroCents } from "../../../../../../definitions/backend/ImportoEuroCents";
import { formatNumberCentsToAmount } from "../../../../../utils/stringBuilder";
import I18n, { setLocale } from "../../../../../i18n";

const fee = 123 as ImportoEuroCents;

describe("PayPalCheckoutPspComponent", () => {
  beforeAll(() => {
    setLocale("it");
  });
  jest.useFakeTimers();
  it(`it should match the snapshot`, () => {
    const component = render(
      <PayPalCheckoutPspComponent
        onEditPress={constNull}
        fee={fee}
        pspName={"pspName"}
        privacyUrl={"https://io.italia.it"}
      />
    );
    expect(component).toMatchSnapshot();
  });
  /*
  TODO uncomment this test when this task is completed https://pagopa.atlassian.net/browse/IA-684?filter=10121
  it(`edit label should be always pressable`, () => {
    const pspName = "pspName123";
    const mockEditLabelOnPress = jest.fn();
    const component = render(
      <PayPalCheckoutPspComponent
        onEditPress={mockEditLabelOnPress}
        fee={fee}
        pspName={pspName}
        privacyUrl={"https://io.italia.it"}
      />
    );
    expect(component).not.toBeNull();

    if (editPspEnabled) {
      const editLabel = component.queryByText(
        I18n.t("global.buttons.edit").toUpperCase()
      );
      expect(editLabel).not.toBeNull();
      fireEvent(editLabel!, "onPress");
      expect(mockEditLabelOnPress).toHaveBeenCalledTimes(1);
    }
  });
   */

  describe("given a fee, pspName and privacy url", () => {
    it(`it should shown the data as expected`, () => {
      const pspName = "pspName123";
      const component = render(
        <PayPalCheckoutPspComponent
          onEditPress={constNull}
          fee={fee}
          pspName={pspName}
          privacyUrl={"https://io.italia.it"}
        />
      );
      expect(component).not.toBeNull();
      expect(
        component.queryByText(formatNumberCentsToAmount(fee, true))
      ).not.toBeNull();
      expect(
        component.queryByText(
          I18n.t("wallet.onboarding.paypal.paymentCheckout.pspHandler", {
            pspName
          })
        )
      ).not.toBeNull();
      expect(
        component.queryByText(
          I18n.t("wallet.onboarding.paypal.paymentCheckout.privacyTerms")
        )
      ).not.toBeNull();
    });
  });

  describe("given a fee, pspName and not privacy url", () => {
    it(`it should not shown the privacy link`, () => {
      const pspName = "pspName123";
      const component = render(
        <PayPalCheckoutPspComponent
          fee={fee}
          pspName={pspName}
          onEditPress={constNull}
        />
      );
      expect(component).not.toBeNull();
      expect(
        component.queryByText(formatNumberCentsToAmount(fee, true))
      ).not.toBeNull();
      expect(
        component.queryByText(
          I18n.t("wallet.onboarding.paypal.paymentCheckout.pspHandler", {
            pspName
          })
        )
      ).not.toBeNull();
      expect(
        component.queryByText(
          I18n.t("wallet.onboarding.paypal.paymentCheckout.privacyTerms")
        )
      ).toBeNull();
    });
  });
});
