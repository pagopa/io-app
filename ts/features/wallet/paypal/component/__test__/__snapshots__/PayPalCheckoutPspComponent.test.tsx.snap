// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayPalCheckoutPspComponent it should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "flex": 1,
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 16,
            "paddingBottom": 13,
            "paddingRight": 16,
            "paddingTop": 13,
          },
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessibilityLabel=""
          accessible={false}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          style={
            Array [
              Object {
                "color": "#475A6D",
                "fontSize": 24,
              },
              Array [
                Object {},
                Object {
                  "color": "#475A6D",
                },
                Object {
                  "fontSize": 24,
                },
              ],
              Object {
                "fontFamily": "io-icon-font",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          hspacer={true}
          small={true}
          style={Object {}}
        />
        <Text
          color="bluegrey"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 18,
            }
          }
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#475A6D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
          weightColorFactory={[Function]}
        >
          Costo della transazione
        </Text>
      </View>
      <View
        spacer={true}
        style={Object {}}
      />
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flex": 1,
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            color="bluegreyDark"
            font="TitilliumWeb"
            fontStyle={
              Object {
                "fontSize": 16,
              }
            }
            style={
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "#17324D",
                  "fontFamily": "Titillium Web",
                  "fontStyle": "normal",
                  "fontWeight": "700",
                },
              ]
            }
            weight="Bold"
            weightColorFactory={[Function]}
          >
            € 1,23
          </Text>
          <Text
            color="bluegrey"
            defaultColor="blue"
            defaultWeight="Bold"
            font="TitilliumWeb"
            fontStyle={
              Object {
                "fontSize": 16,
              }
            }
            style={
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "#475A6D",
                  "fontFamily": "Titillium Web",
                  "fontStyle": "normal",
                  "fontWeight": "400",
                },
              ]
            }
            weight="Regular"
          >
            Gestita da pspName
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    spacer={true}
    style={Object {}}
  />
  <Text
    color="bluegrey"
    defaultColor="blue"
    defaultWeight="Bold"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
      }
    }
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "#475A6D",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Proseguendo dichiari di aver letto e compreso i
  </Text>
  <Text
    accessibilityRole="link"
    color="blue"
    defaultColor="blue"
    defaultWeight="SemiBold"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "textDecorationLine": "underline",
      }
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontSize": 16,
          "textDecorationLine": "underline",
        },
        Object {
          "color": "#0073E6",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "600",
        },
      ]
    }
    weight="SemiBold"
  >
    Termini e condizioni d’uso e l’Informativa Privacy
  </Text>
</View>
`;
